_posts/2015-01-30-cloudpipe.markdown

```diff
@@ -0,0 +1,103 @@
+---
+layout: post
+title: "Put that in your cloud and smoke it"
+date: "2015-01-30"
+category: news
+tags:
+- software
+- go
+- cloudpipe
+image:
+  feature: texture-lakemead-wires.jpg
+---
+
+When you want to execute a long-running or massively parallel process in a persistent way, you have a few options. You can launch a server, install a bunch of software, and stitch things together with `cron`. Or, you can install a [Hadoop](http://hadoop.apache.org/) cluster, rewrite everything from the ground up to use map-reduce or Spark or another powerful concurrency framework, and orchestrate your job with that toolkit.
+
+Those options all have their place, but sometimes, all you want to do is *write some code* and *run it somewhere*. No server spin-up, no system administration, no patches and security updates. [Kyle Kelley](http://lambdaops.com/) and I spent a few weeks around the holidays putting together [an open-source project called **"Cloudpipe"**](https://github.com/cloudpipe/cloudpipe) to accomplish exactly this. Today, we have a proof of concept we can run that successfully executes jobs, and we're hard at work filling in the blanks in the architecture and putting together a real deployment.
+
+## Using it
+
+To start, we chose to target an existing client SDK, Python's [multyvac](https://pypi.python.org/pypi/multyvac/0.5.0), which was nice for a few reasons:
+
+ 1. It gave me a concrete spec to target while I was mapping out the endpoints for the API server. This saved us a lot of bikeshedding in the early days.
+ 2. It let me start backend-first and test things with real code, just as a user would experience them. Otherwise, I would have had to use awkward `curl` scripts until we could design and put together a decent SDK.
+ 3. In theory, people are [already using](https://github.com/search?utf8=%E2%9C%93&q=import+multyvac&type=Code&ref=searchresults) the SDK. (If that search is any indication, though, in retrospect we should have stuck with its predecessor, [cloud](https://github.com/search?utf8=%E2%9C%93&q=import+cloud&type=Code&ref=searchresults), instead.)
+
+As a result, the only client SDK that exists today is for Python (and Python 2, at that). That's going to change, though - we've been planning to [write our own client for Python](https://github.com/cloudpipe/cloudpipe/issues/17) and [Shaunak Kashyap](https://github.com/ycombinator) is already working on [an SDK for node.js](https://github.com/cloudpipe/sdk-node). We're also likely to diverge more and more from multyvac as we move on. Stay tuned for more on that front.
+
+Once you have an account running on a cluster somewhere, you can install multyvac with:
+
+```bash
+pip install multyvac
+```
+
+Then, you can submit functions as jobs:
+
+```python
+import multyvac
+
+# Connect to the cluster
+multyvac.config.set_key(api_key='username',
+                        api_secret_key='54321',
+                        api_url='https://somewhere/v1')
+
+def add(a, b):
+  return a + b
```

Comment:
We really need a better job to submit. I'd use web scraping but then we'd need requests, yada yada.